flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/ExecutionFailureHandlerTest.java

```diff
@@ -171,6 +179,9 @@ void testNonRecoverableFailureHandlingResult() throws Exception {
         assertThat(result.getFailureLabels().get())
                 .isEqualTo(testingFailureEnricher.getFailureLabels());
         assertThat(result.getTimestamp()).isEqualTo(timestamp);
+        // NonRecoverableFailure is new attempt even if RestartBackoffTimeStrategy consider it's not
+        // new attempt.
+        assertThat(result.isNewAttempt()).isTrue();
```

Comment:
```suggestion
        assertThat(result.isNewAttempt())
                .as(
                        "A NonRecoverableFailure should be new attempt even if RestartBackoffTimeStrategy consider it's not new attempt.")
                .isTrue();
```
just another nitty hint: For tests, we don't need to add comments but could rather have a description assertion message. That serves the same purpose as the command but also makes the test execution more descriptive.